### PR TITLE
Shenli/rm rset stmt plan

### PR DESCRIPTION
**Pull request by shenli**
Remove plan, rset and stmt.go. Move interface into ast package.
@coocood

### CODE REVIEW

**Comment by shenli** [comment]
@zimulala

File: util/testkit/testkit.go [hunk 1]
```diff
@@ -100,11 +101,28 @@ func (tk *TestKit) MustQuery(sql string, args ...interface{}) *Result {
 	rs, err := tk.Exec(sql, args...)
 	tk.c.Assert(err, check.IsNil, comment)
 	tk.c.Assert(rs, check.NotNil, comment)
-	rows, err := rs.Rows(-1, 0)
+	rows, err := getRows(rs)
 	tk.c.Assert(err, check.IsNil, comment)
 	return &Result{rows: rows, c: tk.c, comment: comment}
 }
 
+func getRows(rs ast.RecordSet) ([][]interface{}, error) {
```
Comment (coocood) [inline util/testkit/testkit.go]: export this function for tests.

File: tidb_test.go [hunk 2]
```diff
@@ -428,9 +429,26 @@ func matches(c *C, rows [][]interface{}, expected [][]interface{}) {
 	}
 }
 
+func getRows(rs ast.RecordSet) ([][]interface{}, error) {
+	var rows [][]interface{}
+	defer rs.Close()
```
Comment (zimulala) [inline tidb_test.go]: check `rs == nil` ？

File: util/testkit/testkit.go (hunk 1 above)
Comment (zimulala) [inline util/testkit/testkit.go]: Why write the function twice in different package？

**Comment by shenli** [comment]
PTAL 
@coocood @zimulala

File: tidb.go [hunk 3]
```diff
@@ -173,6 +171,27 @@ func runStmt(ctx context.Context, s stmt.Statement, args ...interface{}) (rset.R
 	return rs, errors.Trace(err)
 }
 
+// GetRows gets all the rows from a RecordSet.
+func GetRows(rs ast.RecordSet) ([][]interface{}, error) {
```
Comment (zimulala) [inline tidb.go]: need some tests ？

**Comment by zimulala** [comment]
`field.ResultField` can be removed?

File: tidb.go (hunk 3 above)
Comment (shenli) [inline tidb.go]: It is only used in unit tests.

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**